cmd/crowdsec-cli/support.go

```diff
@@ -0,0 +1,393 @@
+	SUPPORT_CAPI_STATUS_PATH             = "capi_status.txt"
+	SUPPORT_ACQUISITION_CONFIG_BASE_PATH = "config/acquis/"
+	SUPPORT_CROWDSEC_PROFILE_PATH        = "config/profiles.yaml"
+)
+
+func collectMetrics() ([]byte, []byte, error) {
+	log.Info("Collecting prometheus metrics")
+	err := csConfig.LoadPrometheus()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if csConfig.Cscli.PrometheusUrl == "" {
+		log.Warn("No Prometheus URL configured, metrics will not be collected")
+		return nil, nil, fmt.Errorf("prometheus_uri is not set")
+	}
+
+	humanMetrics, err := FormatPrometheusMetric(csConfig.Cscli.PrometheusUrl+"/metrics", "human")
+
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not fetch promtheus metrics: %s", err)
+	}
+
+	req, err := http.NewRequest(http.MethodGet, csConfig.Cscli.PrometheusUrl+"/metrics", nil)
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not create requests to prometheus endpoint: %s", err)
+	}
+	client := &http.Client{}
+	resp, err := client.Do(req)
+
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not get metrics from prometheus endpoint: %s", err)
+	}
+
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not read metrics from prometheus endpoint: %s", err)
+	}
+
+	return humanMetrics, body, nil
+}
+
+func collectVersion() []byte {
+	log.Info("Collecting version")
+	return []byte(cwversion.ShowStr())
+}
+
+func collectOSInfo() ([]byte, error) {
+	log.Info("Collecting OS info")
+	info, err := osinfo.GetOSInfo()
+
+	if err != nil {
+		return nil, err
+	}
+
+	w := bytes.NewBuffer(nil)
+	w.WriteString(fmt.Sprintf("Architecture: %s\n", info.Architecture))
+	w.WriteString(fmt.Sprintf("Family: %s\n", info.Family))
+	w.WriteString(fmt.Sprintf("ID: %s\n", info.ID))
+	w.WriteString(fmt.Sprintf("Name: %s\n", info.Name))
+	w.WriteString(fmt.Sprintf("Codename: %s\n", info.Codename))
+	w.WriteString(fmt.Sprintf("Version: %s\n", info.Version))
+	w.WriteString(fmt.Sprintf("Build: %s\n", info.Build))
+
+	return w.Bytes(), nil
+}
+
+func initHub() error {
+	if err := csConfig.LoadHub(); err != nil {
+		return fmt.Errorf("cannot load hub: %s", err)
+	}
+	if csConfig.Hub == nil {
+		return fmt.Errorf("hub not configured")
+	}
+
+	if err := cwhub.SetHubBranch(); err != nil {
+		return fmt.Errorf("cannot set hub branch: %s", err)
+	}
+
+	if err := cwhub.GetHubIdx(csConfig.Hub); err != nil {
+		return fmt.Errorf("no hub index found: %s", err)
+	}
+	return nil
+}
+
+func collectHubItems(itemType string) []byte {
+	log.Infof("Collecting %s list", itemType)
+	items := ListItems([]string{itemType}, []string{}, false, true, all)
+	return items
+}
+
+func collectBouncers(dbClient *database.Client) ([]byte, error) {
+	return getBouncers(dbClient)
+}
+
+func collectAgents(dbClient *database.Client) ([]byte, error) {
+	return getAgents(dbClient)
+}
+
+func collectAPIStatus(login string, password string, endpoint string, prefix string) []byte {
```

Comment:
This func miss the tls support if I'm not mistaken.
We need to check if the user as setup tls and then try to authenticate and collect.

Reply:
as seen it works oob because magic